flaml/automl/task/time_series_task.py

```diff
@@ -0,0 +1,508 @@
+import logging
+import time
+from typing import List
+
+import pandas as pd
+import numpy as np
+from scipy.sparse import issparse
+from sklearn.model_selection import (
+    GroupKFold,
+    TimeSeriesSplit,
+)
+
+from flaml.automl.ml import get_val_loss, default_cv_score_agg_func
+from flaml.automl.time_series.ts_data import (
+    TimeSeriesDataset,
+    DataTransformerTS,
+    normalize_ts_data,
+)
+
+from flaml.automl.task.task import (
+    Task,
+    get_classification_objective,
+    TS_FORECAST,
+    TS_FORECASTPANEL,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TimeSeriesTask(Task):
```

Comment:
Any specific reason why `TimeSeriesTask` doesn't inherit from `GenericTask`?

Reply:
Because there's no opportunity for code reuse. Only stuff that's task-specific belongs into the ***Task classes, so what's the point of inheritance?

Reply:
Well, maybe we'd eventually want subclasses of eg TimeSeriesTask specific to different forecasting tasks, but certainly GenericTask is on a different branch of the hierarchy